assets/src/components/activities/ordering/OrderingDelivery.tsx

```diff
@@ -0,0 +1,210 @@
+    props.onSubmitActivity(attemptState.attemptGuid,
+      // update this input too
+      [{ attemptGuid: attemptState.parts[0].attemptGuid, response: { input: orderedChoiceIds() } }])
+      .then((response: EvaluationResponse) => {
+        if (response.evaluations.length > 0) {
+          const { score, out_of, feedback, error } = response.evaluations[0];
+          const parts = [Object.assign({}, attemptState.parts[0], { feedback, error })];
+          const updated = Object.assign({}, attemptState, { score, outOf: out_of, parts });
+          setAttemptState(updated);
+        }
+      });
+  };
+
+  const updateSelection = (id: string) => {
+    const newSelection = !!selected.find(s => s === id)
+      ? selected.filter(s => s !== id)
+      : selected.concat([id]);
+    setSelected(newSelection);
+  };
+
+  const onSelect = (id: string) => {
+    // Update local state by adding or removing the id
+    updateSelection(id);
+
+    // Post the student response to save it
+    // Here we will make a list of the selected ids like { input: [id1, id2, id3].join(' ')}
+    // Then in the rule evaluator, we will say
+    // `input like id1 && input like id2 && input like id3`
+    props.onSaveActivity(attemptState.attemptGuid,
+      [{ attemptGuid: attemptState.parts[0].attemptGuid,
+        response: { input: orderedChoiceIds() } }]);
+  };
+
+  const onRequestHint = () => {
+    props.onRequestHint(attemptState.attemptGuid, attemptState.parts[0].attemptGuid)
+    .then((state: RequestHintResponse) => {
+      if (state.hint !== undefined) {
+        setHints([...hints, state.hint] as any);
+      }
+      setHasMoreHints(state.hasMoreHints);
+    });
+  };
+
+  const onReset = () => {
+    props.onResetActivity(attemptState.attemptGuid)
+    .then((state: ResetActivityResponse) => {
+      setSelected([]);
+      setAttemptState(state.attemptState);
+      setModel(state.model as OrderingModelSchema);
+      setHints([]);
+      setHasMoreHints(props.state.parts[0].hasMoreHints);
+    });
+  };
+
+  const evaluationSummary = isEvaluated
+    ? <Evaluation key="evaluation" attemptState={attemptState}/>
+    : null;
+
+  const reset = isEvaluated && !props.graded
+    ? (<div className="d-flex my-3">
+        <div className="flex-fill"></div>
+        <Reset hasMoreAttempts={attemptState.hasMoreAttempts} onClick={onReset} />
+      </div>
+    )
+    : null;
+
+  const ungradedDetails = props.graded ? null : [
+    evaluationSummary,
+    <Hints key="hints" onClick={onRequestHint} hints={hints}
+      hasMoreHints={hasMoreHints} isEvaluated={isEvaluated}/>];
+
+  const maybeSubmitButton = props.graded
+    ? null
+    : (
+      <button
+        className="btn btn-primary mt-2 float-right"
+        disabled={isEvaluated || selected.length !== choices.length}
+        onClick={onSubmit}>
+        Submit
+      </button>
+    );
+
+  return (
+    <div className={`activity multiple-choice-activity ${isEvaluated ? 'evaluated' : ''}`}>
```

Comment:
multiple-choice-activity reference here should be changed to be ordering activity specific